### DISCUSSION TITLE
Use .editorconfig by default

### DISCUSSION THREAD

**Discussion by florianrusch-zf** [eclipse-edc/Connector#700] [Ideas]
Wouldn't it make sense to move the `.editorconfig` to the root directory of the project so that it is used by default? If the user does not want to use it, he can configure his IDE not to use it.

Furthermore I noticed that the settings in the files `.editorconfig` and `edc-checkstyle.xml` do not match. The imports are sorted differently.

**Comment by c-r-b-9** [comment]
After having a look at this: I'm not sure maintaining two separate code style configurations is beneficial. We can consolidate this now, but we'll have to keep them synced. Since pretty much any IDE can do checkstyle, I'd get rid of the .editorconfig entirely. What do you think?

**Comment by paullatzelsperger** [comment]
What advantages would the `.editorconfig` bring over the current setup (cf. `styleguide.md`)? What problems does it solve, that the current setup cannot? 

On a general scope though topics like this should be a discussion first. Issues should be created only after a decision is made and a clear action is identified. will convert.

**Reply by ndr-brt** [reply]
In fact, there's an editorconfig file in the repository (`edc-codestyle.editorconfig`), I think that's where the confusion pops up.
If that's not needed, we should remove it and rely only on the checkstyle configuration.

**Reply by paullatzelsperger** [reply]
Ah, I recall now. this was intended as a fallback. if no one's using it, I'm all for deleting it. If people are in fact using it, it needs to be synched with the checkstyle config.

**Reply by florianrusch-zf** [reply]
Thank you, @paullatzelsperger, for moving it into a discussion! It's definitely better addressed here. I'm also in favor of removing it.

Additionally, we can make the description of how to setup e.g. IntelliJ more precise. For me, who has never worked with checkstyle, it was difficult to identify what I have to do in my IDE.

**Comment by florianrusch-zf** [comment]
It seems that the editorconfig and the checkstyle definition are not really different from each other. I had a bug in my IDE where the imports layout was not applied appropriately 😣  So please don't worry about it anymore.
